src/iinit.cpp

```diff
@@ -66,11 +69,225 @@ mkrodsdir() {
+                fmt::print(
+                    stderr,
+                    "Failed to parse environment file: []\n"
+                    "Falling back to original environment settings.",
+                    e.what());
+            }
+
+            obj_to_dump.merge_patch(_update);
+        }
+        else {
+            obj_to_dump = _update;
+        }
+
+        if (std::ofstream f(env_file.c_str(), std::ios::out); f.is_open()) {
+            f << obj_to_dump.dump(4) << "\n";
+            f.close();
+        }
+        else {
+            fmt::print("failed to open environment file [{}]\n", env_file);
+        }
+    } // save_updates_to_irods_environment
+
+    auto option_specified(std::string_view _option, int argc, char** argv) -> bool
+    {
+        for (int arg = 0; arg < argc; ++arg) {
+            if (!argv[arg]) {
+                continue;
+            }
+
+            if (_option == argv[arg]) {
+                // parseCmdLineOpt is EVIL and requires this. Please don't ask why.
+                argv[arg] = "-Z";
+                return true;
+            }
+        }
+
+        return false;
+    } // option_specified
+
+    auto set_env_from_prompt(char _setting[], const char* _prompt, std::size_t _len) -> void
+    {
+        // If the setting already has a value, use that as the default value and show it in the prompt.
+        if (0 == std::strlen(_setting)) {
+            fmt::print("{}: ", _prompt);
+        }
+        else {
+            fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), _setting);
+        }
+
+        std::string response;
+        std::getline(std::cin, response);
+
+        // If the response is empty, don't copy anything. If there is a value in _setting already, it will be used
+        // as the default response. If there is nothing, then there is nothing to copy anyway.
+        if (!response.empty()) {
+            std::strncpy(_setting, response.c_str(), _len);
+        }
+    } // set_env_from_prompt
+
+    auto set_env_from_prompt(char _setting[], const char* _default, const char* _prompt, std::size_t _len) -> void
+    {
+        const bool env_has_value = 0 != std::strlen(_setting);
+        const char* default_value = env_has_value ? _setting : _default;
+
+        fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), default_value);
+        std::string response;
+        std::getline(std::cin, response);
+
+        // The else case indicates that the response was empty and the environment already has a value. If the
+        // environment has a value already, the default value is the existing value (meaning, there will be no
+        // change in the environment). Therefore, no copy occurs in the else case as no change will occur.
+        if (!response.empty()) {
+            std::strncpy(_setting, response.c_str(), _len);
+        }
+        else if (!env_has_value) {
+            std::strncpy(_setting, default_value, _len);
+        }
```

Comment:
I feel the comment doesn't quite reflect the code...  please read again and confirm/deny.

Reply:
Oops, this didn't appear in the Files Changed tab.

I think the comment reflects the code. There is no else case and it is explaining what the else case would mean if there was code here to represent it. There is no code to represent it because it's a no-op.